module-2/backend/week_two.md

```diff
@@ -4,29 +4,43 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 
 Note: When you're done, submit a PR. 
 
-1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+1. At a high level, what is ActiveRecord? What does it do/allow you to do? 
+  It is a Ruby Library that allows you to work with relational SQL/
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+  They are associations (would they be considered class methods? An example would be that a car belongs to one driver but a     driver could have many cars.
 3. What do they allow you to do?
+  It allows you to call on the relationship between to tables and call methods on them to find the different objects from both   ends of the relationship
 4. What's the difference between agile workflow and waterfall method?
+  Waterfall is a very incremental form of development Where you would develop, then test, then fix any errors, then finalize     for customers, then release. In an Agile workflow is a more iteration driven approach where they get a single functionality   created and tested, prepared and launched and then move onto the next iteration.
 5. What is the difference between `#new` and `#create`?
+  New doesnt save the new instance of the object it only makes it. Create makes the new object AND saves it
 6. At a basic level, what does cURL allow you to do?
+  it allows you to send information using URL syntax?
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
-8. Define foreign key, primary key, and schema.
+ Students have many teachers and teachers have many students.
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+  A foreign key is a reference in a table to the primary key in another table. It is how the key with the foreign key has       access to the information in the other table where the primary key is.
 10. What are the parts of an HTTP response?
+    status line, message, body
 11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+  I'm not quite sure I know/remember what you mean by DRY...
```

Comment:
DRY => Don't Repeat Yourself. Partials might be one example since they help us avoid putting code that is basically the same in two different places.